### DISCUSSION TITLE
How to use laravel-permission with API versioning model types

### DISCUSSION THREAD

**Discussion by BlumDev92** [spatie/laravel-permission#2012] [Q&A] [answered]
Hello there,

I implemented a V2 of my API which uses a different user model. My previous User model had the model_type `App\Models\Base\User`. But right now for V2 the namespace is `App\Models\V2\Base\User`.
This change is breaking my current setup, because all roles and permissions use  `App\Models\Base\User`.

Is it possible to use something like a middleware to modify the model_type if the user is accessing the API via a v2 endpoint?

Thank you for your help.

**Comment by erikn69** [comment, marked as answer]
use `Relation::morphMap` to alias `model_type` to a standard  name
[enforcing-morph-maps-in-laravel](https://laravel-news.com/enforcing-morph-maps-in-laravel)
[custom-polymorphic-types](https://laravel.com/docs/master/eloquent-relationships#custom-polymorphic-types)

**Reply by BlumDev92** [reply]
Thanks for your answer 💯 . Where do I have to put `Relation::morphMap`? Because this should only apply if the user is accessing v2 endpoints.

**Reply by erikn69** [reply]
Many ways, depends on you
What I would do is add a middleware on routes for use v1 and v2 model with the same alias but with different paths to user model file

**Reply by BlumDev92** [reply]
I added a APIVersion Middleware like this:

```
$apiVersion = $guard == '' ? '' : ('\\' . strtoupper($guard) . '\\');
Relation::morphMap([
    'App\Models\Base\User' => 'App\Models' . $apiVersion . 'Base\User'
]);
```

works like a charm. Thank you.
